siliconcompiler/schema/schema_cfg.py

```diff
@@ -215,6 +215,74 @@ def schema_cfg():
     # Packaging
     cfg = schema_package(cfg)
 
+    # Packaging
+    cfg = schema_schematic(cfg)
+
+    return cfg
+
+
+###############################################################################
+# SCHEMATIC
+###############################################################################
+def schema_schematic(cfg):
+    ''' Schematic
+    '''
+
+    scparam(cfg, ['schematic', 'component'],
+            sctype='[(str,str)]',
+            shorthelp="Schematic: components",
+            switch="-schematic_component <(str,str)>",
+            example=["cli: -schematic_component (B0,NAND2X1)",
+                     "api: chip.set('schematic', 'component', ('B0','NAND2X1'))"],
+            schelp="""Components in the design, specified as a list of
+            (INSTANCE,MODEL) tuples.""")
+
+    scparam(cfg, ['schematic', 'pin'],
+            sctype='[(str,str)]',
+            shorthelp="Schematic: pins",
+            switch="-schematic_pin 'name <(str,str)>'",
+            example=["cli: -schematic_pin '(A,input)",
+                     "api: chip.set('schematic', 'pin', ('A','input'))"],
+            schelp="""Primary design I/O pins specified as a list of
+            (NAME,DIRECTION) tuples. Name can be any legal verilog string
+            name while DIRECTION is limited to input, output, and inout.""")
```

Comment:
why not make this
`schematic`, `pin`, name, `direction`, `enum`? this way the schema can enforce the legal values

Reply:
Fixed and committed.